### PR TITLE
Migrate from dbus-glib to gdbus

### PR DESCRIPTION
**Pull request by yetist**
Please review, thank you.

### CODE REVIEW

**Comment by lukefromdc** [comment]
Is this a duplicate of https://github.com/mate-desktop/mate-session-manager/pull/170 or should that PR be closed in favor of this one?

**Comment by yetist** [comment]
Both #170 and #171 are migrate dbus-glib to gdbus, but different files are changed.
Maybe I should merge them into one PR?

**Comment by sc0w** [comment]
@yetist It's your decision, but if you like to split, I think the files affected need to mentioned in the commit message and the title of PR

**Comment by raveit65** [comment]
Only rewording the commit and adding the affected files to commit body is needed from my point of view.
In both PR's. Please.....

**Comment by lukefromdc** [comment]
I do not know enough about GError to know whether  g_error_free only needs to be run when the GError variable has been set to a non-NULL value or if it always needs to  be run. Other than that memory management is probably OK. I don't know enough about either dbus system to spot any screwup in the use of gdbus, but it looks like a straight code replacement

**Comment by lukefromdc** [comment]
Session saving also works fine, so no issues there

**Comment by lukefromdc** [comment]
Should I merge this? Nobody has done anything further with it in a week

**Comment by monsta** [comment]
I can't figure out how to use this test app...

File: mate-session/test-client-dbus.c [hunk 1]
```diff
@@ -64,129 +58,138 @@ session_manager_connect (void)
...
+        res = g_dbus_proxy_call_sync (proxy,
+                                      "EndSessionResponse",
+                                      g_variant_new ("(bs)",
+                                                     is_ok,
+                                                     reason),
+                                      G_DBUS_CALL_FLAGS_NONE,
+                                      -1,
+                                      NULL,
+                                      &error);
+        if (error != NULL) {
```
Comment (monsta) [inline mate-session/test-client-dbus.c]: Does `g_dbus_proxy_call_sync` set it to `NULL` on success, or just doesn't touch it in that case?
It's probably better to check the return value, like it's done in #170.

Comment (monsta) [reply mate-session/test-client-dbus.c]: Same in two similar situations below

**Comment by monsta** [comment]
Thanks. I'll squash + merge it as it's needed to have a working test app for #170.

**Pull request merged**